### ISSUE TITLE
Some rules fail with NPE on traits

### ISSUE THREAD

**Issue by topr** [CodeNarc/CodeNarc#93]
At least for the following rules:
- `BracesForClass`
- `SpaceAfterOpeningBrace`
- `SpaceBeforeOpeningBrace`

CodeNarc throws `NullPointerException` if applied on a Groovy `trait`. I guess the support for traits it's not implemented yet as it's quite new language feature. Although other rules I'm using work al lright which makes me think it shouldn't be to big chunk of work to do (how do you think?).

I'm running on:
- Groovy 2.4.3
- CodeNarc 0.23
- Gradle 2.4

Both Gradle 2.4 and 2.3 give the same error although with Gradle 2.3 stacktrace is printed:
```
$ ./gradlew clean codenarcMain
:buildSrc:compileJava UP-TO-DATE
:buildSrc:compileGroovy UP-TO-DATE
:buildSrc:processResources UP-TO-DATE
:buildSrc:classes UP-TO-DATE
:buildSrc:jar UP-TO-DATE
:buildSrc:assemble UP-TO-DATE
:buildSrc:compileTestJava UP-TO-DATE
:buildSrc:compileTestGroovy UP-TO-DATE
:buildSrc:processTestResources UP-TO-DATE
:buildSrc:testClasses UP-TO-DATE
:buildSrc:test UP-TO-DATE
:buildSrc:check UP-TO-DATE
:buildSrc:build UP-TO-DATE
:yolo:clean
:yolo:codenarcMainError from [org.codenarc.rule.formatting.SpaceBeforeOpeningBraceRule] processing source file [C:/dev/proj/yolo/src/main/groovy/AnyTrait.groovy]
java.lang.NullPointerException
	at org.codenarc.util.AstUtil.findFirstNonAnnotationLine(AstUtil.java:1085)
	at org.codenarc.rule.AbstractAstVisitor.sourceLine(AbstractAstVisitor.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:163)
	at org.codenarc.rule.formatting.AbstractSpaceAroundBraceAstVisitor.sourceLineOrEmpty(AbstractSpaceAroundBraceAstVisitor.groovy:43)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:163)
	at org.codenarc.rule.formatting.SpaceBeforeOpeningBraceAstVisitor.visitClassEx(SpaceBeforeOpeningBraceRule.groovy:47)
	at org.codenarc.rule.AbstractAstVisitor.visitClass(AbstractAstVisitor.java:147)
	at org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
	at org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:141)
	at org.codenarc.rule.Rule$applyTo$1.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codenarc.rule.Rule$applyTo$1.call(Unknown Source)
	at org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
	at org.codenarc.ant.AntFileSetSourceAnalyzer.processFile(AntFileSetSourceAnalyzer.java:168)
	at org.codenarc.ant.AntFileSetSourceAnalyzer.access$0(AntFileSetSourceAnalyzer.java:165)
	at org.codenarc.ant.AntFileSetSourceAnalyzer$1.run(AntFileSetSourceAnalyzer.java:157)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
```

**Comment by mkordas** [comment]
+1

**Comment by chrismair** [comment]
Fixed by #108
